source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/PropertyContactEditForm.tsx

```diff
@@ -0,0 +1,142 @@
+import { Formik, FormikProps } from 'formik';
+import React, { useMemo, useState } from 'react';
+import styled from 'styled-components';
+
+import { Input, Select, SelectOption, TextArea } from '@/components/common/form';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { StyledSummarySection } from '@/components/common/Section/SectionStyles';
+import { useOrganizationRepository } from '@/features/contacts/repositories/useOrganizationRepository';
+import { IContactSearchResult } from '@/interfaces';
+import { Api_OrganizationPerson } from '@/models/api/Organization';
+import { Api_PropertyContact } from '@/models/api/Property';
+import { formatApiPersonNames, formatContactSearchResult } from '@/utils/personUtils';
+
+import { PropertyContactFormModel } from './models';
+import { PropertyContactEditFormYupSchema } from './validation';
+
+export interface IPropertyContactEditFormProps {
+  isLoading: boolean;
+  propertyContact: Api_PropertyContact;
+  onSave: (apiModel: Api_PropertyContact) => void;
+}
+
+export const PropertyContactEditForm = React.forwardRef<
+  FormikProps<any>,
+  IPropertyContactEditFormProps
+>((props, ref) => {
+  const {
+    getOrganizationDetail: {
+      execute: fetchOrganization,
+      response: organization,
+      loading: getOrganizationLoading,
+    },
+  } = useOrganizationRepository();
+
+  const [contact, setContact] = useState<IContactSearchResult | null>(null);
```

Comment:
I'm not a big fan of this and the repo call being in this form component. Container components should be responsible for state and data fetching as per our best practices and for ease of testing.

Reply:
I agree. However we need a organization primary contact component that takes care of that abstraction similar to the contact picker